### PR TITLE
Dockerfile for CentOS CI build

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

Dockerfile for CentOS CI build

**Motivation and Context**
Setting up CI build for CentOS

To use it:
```bash
mkdir /data/b
tools/ci_build/github/linux/run_dockerbuild.sh -p 3.6 -o centos7 -d cpu -r /data/b -x "--use_automl --build_wheel"
```

- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: tools/ci_build/github/linux/docker/scripts/install_centos.sh [hunk 1]
```diff
@@ -1,12 +1,7 @@
...
-yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm https://centos7.iuscommunity.org/ius-release.rpm
+yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
 
-yum remove --tolerant cmake git
-yum install -y cmake3 git2u-all
-ln -s /usr/bin/cmake3 /usr/bin/cmake
-ln -s /usr/bin/ctest3 /usr/bin/ctest
-
-yum install -y redhat-lsb-core expat-devel libcurl-devel protobuf-compiler protobuf-devel protobuf rpmdevtools tar unzip ccache curl gcc gcc-c++ zlib-devel make  python2-devel python3-devel python3-numpy libunwind icu aria2 rsync python3-setuptools python3-wheel bzip2
+yum install -y redhat-lsb-core expat-devel libcurl-devel rpmdevtools tar unzip ccache curl gcc gcc-c++ zlib-devel make  python2-devel python3-devel python3-pip python3-numpy libunwind icu aria2 rsync python3-setuptools python3-wheel bzip2 git
```
Comment (yuslepukhin) [inline tools/ci_build/github/linux/docker/scripts/install_centos.sh]: > git [](start = 238, length = 4)

We don't want this git. It is too old

File: tools/ci_build/github/linux/docker/scripts/install_deps.sh [hunk 2]
```diff
@@ -1,23 +1,53 @@
...
   cp /tmp/azcopy/azcopy /usr/bin
+  echo "Installing cmake"
+  aria2c -q -d /tmp/src "https://github.com/Kitware/CMake/releases/download/v3.15.4/cmake-3.15.4-Linux-x86_64.tar.gz"
+  tar -zxf /tmp/src/cmake-3.15.4-Linux-x86_64.tar.gz --strip=1 -C /usr
+else
+  echo "Installing cmake"
+  aria2c -q -d /tmp/src https://github.com/Kitware/CMake/releases/download/v3.15.4/cmake-3.15.4.tar.gz
+  tar -xf /tmp/src/cmake-3.15.4.tar.gz -C /tmp/src
+  cd /tmp/src/cmake-3.15.4
+  ./configure --prefix=/usr --parallel=`nproc` --system-curl --system-zlib --system-expat
```
Comment (yuslepukhin) [inline tools/ci_build/github/linux/docker/scripts/install_deps.sh]: >./configure --prefix=/usr --parallel=`nproc` [](start = 2, length = 44)

I was wondering why do we need this?

File: tools/ci_build/github/linux/docker/scripts/install_centos.sh [hunk 3]
```diff
@@ -1,12 +1,7 @@
 #!/bin/bash
 set -e
 
-yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm https://centos7.iuscommunity.org/ius-release.rpm
+yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
```
Comment (yuslepukhin) [inline tools/ci_build/github/linux/docker/scripts/install_centos.sh]: >yum install [](start = 0, length = 11)

I suggest we refersh the repo with yum repolist

File: tools/ci_build/github/linux/docker/scripts/install_centos.sh (hunk 1 above)
Comment (snnn) [reply tools/ci_build/github/linux/docker/scripts/install_centos.sh]: But it's good enough to running the build.

Comment (yuslepukhin) [reply tools/ci_build/github/linux/docker/scripts/install_centos.sh]: But it is NOT good enough if we want to login and do rudimentary things.

---
In reply to: [331164812](https://github.com/microsoft/onnxruntime/pull/1986#discussion_r331164812) [](ancestors = 331164812)

File: tools/ci_build/github/linux/docker/scripts/install_deps.sh (hunk 2 above)
Comment (snnn) [reply tools/ci_build/github/linux/docker/scripts/install_deps.sh]: For x86_32, cmake doesn't provide a prebuilt binary.

Comment (yuslepukhin) [reply tools/ci_build/github/linux/docker/scripts/install_deps.sh]: They are probably right.

---
In reply to: [331165249](https://github.com/microsoft/onnxruntime/pull/1986#discussion_r331165249) [](ancestors = 331165249)

File: tools/ci_build/github/linux/docker/scripts/install_centos.sh (hunk 1 above)
Comment (snnn) [reply tools/ci_build/github/linux/docker/scripts/install_centos.sh]: Then, may I build the git from source? I don't quite trust third-party yum repos.

Comment (yuslepukhin) [reply tools/ci_build/github/linux/docker/scripts/install_centos.sh]: And how does it fix it, building from their source?

---
In reply to: [331166421](https://github.com/microsoft/onnxruntime/pull/1986#discussion_r331166421) [](ancestors = 331166421)

**Comment by snnn** [comment]
Will fix the comments in the next PR.

**Pull request merged**